File: android-beans/src/test/java/rocks/frieler/android/beans/scopes/activity/ActivityScopedBeanHolderTest.java
package rocks.frieler.android.beans.scopes.activity;

import org.junit.Test;
import org.junit.runner.RunWith;
import org.mockito.junit.MockitoJUnitRunner;

import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.verify;
import static org.mockito.Mockito.verifyZeroInteractions;

@RunWith(MockitoJUnitRunner.class)
public class ActivityScopedBeanHolderTest {
    private ActivityScopedFactoryBeanHandler.ActivityScopedBeanHolder<Object> beanHolder
            = new ActivityScopedFactoryBeanHandler.ActivityScopedBeanHolder<>();

    @Test
    public void testOnClearedIgnoresActivityUnawareBean() {
        Object bean = mock(Object.class);
        beanHolder.setBean(bean);

        beanHolder.onCleared();

        verifyZeroInteractions(bean);
    }

    @Test
    public void testOnClearedClearsActivityAwareBean() {
        ActivityAware bean = mock(ActivityAware.class);
        beanHolder.setBean(bean);

        beanHolder.onCleared();

        verify(bean).setActivity(null);
    }
}

File: android-beans/src/test/kotlin/rocks/frieler/android/beans/scopes/activity/ActivityScopedBeanHolderTest.kt
package rocks.frieler.android.beans.scopes.activity

import org.junit.jupiter.api.Test
import org.mockito.kotlin.mock
import org.mockito.kotlin.verify
import org.mockito.kotlin.verifyNoInteractions
import rocks.frieler.android.beans.scopes.activity.ActivityScopedFactoryBeanHandler.ActivityScopedBeanHolder

class ActivityScopedBeanHolderTest {
	private val beanHolder = ActivityScopedBeanHolder<Any>()

	@Test
	fun `onCleared() ignores Activity-unaware bean`() {
		val bean : Any = mock()

		beanHolder.bean = bean
		beanHolder.onCleared()

		verifyNoInteractions(bean)
	}

	@Test
	fun `onCleared() clears ActivityAware bean`() {
		val bean : ActivityAware = mock()

		beanHolder.bean = bean
		beanHolder.onCleared()

		verify(bean).setActivity(null)
	}
}
